execinfo.h is part of glibc and is used in crash reporting to show
backtraces. Alpine Linux used to have libexecinfo package, but that was
removed at some point. So the only fix now is to disable inclusion of
execinfo.h. This is what e.g. godot package is also doing.
diff --git a/AK/Assertions.cpp b/AK/Assertions.cpp
index 57ea335a..4a0b4927 100644
--- a/AK/Assertions.cpp
+++ b/AK/Assertions.cpp
@@ -10,10 +10,6 @@
 #include <AK/StringBuilder.h>
 #include <AK/StringView.h>
 
-#if (defined(AK_OS_LINUX) && !defined(AK_OS_ANDROID)) || defined(AK_LIBC_GLIBC) || defined(AK_OS_BSD_GENERIC) || defined(AK_OS_SOLARIS) || defined(AK_OS_HAIKU)
-#    define EXECINFO_BACKTRACE
-#endif
-
 #if defined(AK_OS_ANDROID) && (__ANDROID_API__ >= 33)
 #    include <android/log.h>
 #    define EXECINFO_BACKTRACE
